packages/utils/interactive-hooks/src/useAsToggleWithEvent.ts

```diff
@@ -24,11 +25,14 @@ export function useAsToggleWithEvent(
 
   const onChange = React.useCallback(
     (e: any) => {
+      if (Platform.OS === 'android') {
+        LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
+      }
```

Comment:
Is this applicable to everything that uses this hook?

Reply:
@Saadnajmi currently this is called only for two components , Switch and Checkbox , we have a clear requirement and working as expected.   
On checkbox still design discussion planned soon , but seems checkbox will not have layout animations rather it can have ripple animations but this doesn't affect it's layout  , but this seemed to best to have according to the RN Core docs.

Reply:
This isn't where I would expect to find this code. I would expect that visual changes would be localized to the component that needs them, and that this hook focuses solely on executing the toggle callback.

I think it would be better to pass in the animation code as part of "userCallback"

Reply:
Hmm that's a good idea , let me try that once.

Reply:
Changes done and approved.